spec/models/registration_system_spec.rb

```diff
@@ -105,4 +107,134 @@
       expect(RegistrationSystem.verify_credentials).to be_truthy
     end
   end
+
+  describe ".update_rhsm_conf_queue" do
+    before do
+      EvmSpecHelper.create_guid_miq_server_zone
+      @proxy_args = {:registration_http_proxy_server   => "192.0.2.0:myport",
+                     :registration_http_proxy_username => "my_dummy_username",
+                     :registration_http_proxy_password => "my_dummy_password"}
+
+      @miq_region = FactoryGirl.create(:miq_region, :region => 1)
+      allow(MiqRegion).to receive(:my_region).and_return(@miq_region)
+    end
+
+    it "does not modify original arguments" do
+      cloned_creds = @proxy_args.clone
```

Comment:
I wonder if we should just freeze the hash and assert we don't get `RuntimeError: can't modify frozen Hash`

I ask this since freeze is for preventing modifications... which is what you're asserting isn't happening.